third_party/nvfuser/csrc/scheduler/reduction_utils.h

```diff
@@ -29,6 +29,38 @@ TORCH_CUDA_CU_API void multiReductionInliner(
     std::vector<std::pair<TensorView*, TensorView*>> cached_outputs,
     std::vector<TensorView*> dummy_outputs = {});
 
+// Propagate transformations with inernal cutoff boundary at boundaryNodesSet
+// in P2C forward propagate, disable propagation to TensorView in
+// boundaryNodesSet in C2P backward propagate, disable propagation from
+// TensorView in boundaryNodesSet
+TORCH_CUDA_CU_API void propagateTransformation(
+    TensorView* reference_tv,
+    const std::unordered_set<TensorView*>& boundaryNodesSet);
+
+// Propagate RFactor from first reduction TensorView to others
+TORCH_CUDA_CU_API void propagateRFactor(
+    TensorView* reference_tv,
+    std::vector<TensorView*> reduction_tvs);
+
+// Propagate Parallelization from reference TensorView to others
+TORCH_CUDA_CU_API void propagateParallelization(
+    Fusion* fusion,
+    TensorView* reduction_tv,
+    TensorView* reference_tv,
+    const bool unroll,
+    const bool vectorize,
+    std::vector<TensorView*> cached_inputs,
+    std::vector<std::pair<TensorView*, TensorView*>> cached_outputs);
+
+// We use axes ignoring broadcasts because in checkPatternEquivalence,
+// broadcast is ignored, we might end up having multiple reductions with
+// pattern equivalence but have different number of broadcasts, so the
+// position in the reference tensor is not necessary the same as the
+// position in other reduction TVs.
```

Comment:
This comment doesn't seem to say what this function does.

Reply:
you're right. Removed from header.